processor/src/test/java/org/mapstruct/ap/test/nestedtargetproperties/ChartEntryToArtist.java

```diff
@@ -13,14 +13,15 @@
 import org.mapstruct.Mapper;
 import org.mapstruct.Mapping;
 import org.mapstruct.Mappings;
+import org.mapstruct.NullValuePropertyMappingStrategy;
 import org.mapstruct.ap.test.nestedsourceproperties._target.ChartEntry;
 import org.mapstruct.ap.test.nestedsourceproperties.source.Chart;
 import org.mapstruct.factory.Mappers;
 
 /**
  * @author Sjaak Derksen
  */
-@Mapper
+@Mapper( nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.SET_TO_NULL)
```

Comment:
This is strange as well. I expected only update methods were adapted.. Need to checkout why I need a setting here..

Reply:
Explanation: update methods are generated to cope with the nested mappings. One of such methods is `chartEntryToSong2`. The setting is needed to get exactly the same behaviour as before. So.. Perhaps `NullValuePropertyMappingStrategy.SET_TO_NULL` would make a better default. Not because of logical reasoning, but just to remain backward compatible.

Reply:
In this case a force set to `NullValuePropertyMappingStrategy.SET_TO_NULL` will actually be the most correct solution and would work nicely for the generated method

Reply:
That means that I need to adapt the testcase..

Reply:
I think that this is acceptable